app/Http/Controllers/Stripe/WebhookController.php

```
@@ -0,0 +1,228 @@
+            $invoice = $client->invoices->retrieve($invoiceId, []);
+        } catch (\Throwable $e) {
+            report($e);
+
+            return; // skip when invoice retrieval fails
+        }
+        $subscriptionId = (string) ($invoice->subscription ?? '');
+        $customerId = (string) ($invoice->customer ?? '');
+        if ($subscriptionId !== '' && $customerId !== '') {
+            $this->syncSubscriptionById($subscriptionId, $customerId, forcePaid: true);
+        }
+    }
```

PHPMD (2.15.0)

Avoid unused private methods such as 'syncSubscriptionByInvoiceId'. (undefined)

(UnusedPrivateMethod)